### PR TITLE
Add Codecov

### PR DESCRIPTION
**Pull request by lucacome**
- Added code coverage report
- Added upload to codecov.io

### CODE REVIEW

**Comment by tomasohaodha** [comment]
Is this uploading our code coverage to a 3rd party site? Is this something we want to do? What does it get us?

**Comment by tomasohaodha** [comment]
@vepatel would you mind commenting here on this one?

**Review by ciarams87 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -134,7 +134,12 @@ jobs:
         with:
           go-version: ${{ needs.checks.outputs.go_version }}
       - name: Run Tests
-        run: go test ./...
+        run: make cover
+      - name: Upload coverage to Codecov
```
Comment (soneillf5) [inline .github/workflows/ci.yml:138]: why are we uploading our coverage to Codecov?

Comment (lucacome) [reply .github/workflows/ci.yml:138]: to get the report (the first comment in the PR) and I guess to have a history to compare to for the coverage diff

Comment (soneillf5) [reply .github/workflows/ci.yml:138]: Sorry, I guess I should be more clear. Why do we upload our code coverage data to a 3rd party website ?
Also, the description of you PR is very minimal, it states what you have done, not why.

Comment (lucacome) [reply .github/workflows/ci.yml:138]: I'm not sure what more there is to explain. Codecov is a tool that shows the coverage for the PR and how it changes from the baseline. In order to have this information in the PR (and for the codebase), you need to upload the data to their website to be analyzed. You can find out more about it here https://about.codecov.io/for/open-source/

What is the concern with uploading the coverage result to Codecov?

By the way, this process is similar for other tools we use like Fossa and Code Scanning in GitHub.

**Review by vepatel — APPROVED** [review]

**Pull request merged**